### PR TITLE
Enable TLS and cert-manager created certs for helm chart

### PR DESCRIPTION
**Pull request by dbaker-rh**
See also #710 -- this is a similar change to the helm chart:
1/ Enable TLS for worker/master communications
2/ Optionally use cert-manager to bootstrap CA, and then create secrets
3/ Update grpc_health_check call to use the TLS certs if enabled

### CODE REVIEW

**Review by marquiz — COMMENTED** [review]
Thanks @dbaker-rh for working on this!

We need to enable certs for topology-updater (`deployment/helm/node-feature-discovery/templates/topologyupdater.yam`). It might make sense to pack all the (non-ca) certs in one yaml file(?)

Also, could you update documentation at `docs/get-started/deployment-and-usage.md`. At least document the new parameters added to values.yaml.

**Comment by dbaker-rh** [comment]
Great comments.  Thank you, Markus.  Just as TLS got forgotten in the grpc update, I forgot topology updated because I don't use it.  I'll update my PR with those changes shortly.

**Comment by dbaker-rh** [comment]
Re: the docs, it would also make sense to provide the full openssl command sequences needed to manually create the TLS certs that nfd will use.  I've got edits to the one document across the two open PRs (kustomize + helm) so I don't want to add that into the mix at the moment, but I'll put it together shortly for completeness.

File: deployment/helm/node-feature-discovery/templates/topologyupdater.yaml [hunk 1]
```diff
@@ -65,20 +76,29 @@ spec:
       - name: host-sys
         hostPath:
           path: "/sys"
+          type: Directory
```
Comment (marquiz) [inline deployment/helm/node-feature-discovery/templates/topologyupdater.yaml:79]: nit: unrelated. For completeness/consistency it would make sense to add the type field to all `hostPath` mounts in all templates. Maybe you could submit a separate PR about this?

File: deployment/helm/node-feature-discovery/templates/cert-manager-certs.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+spec:
+  secretName: nfd-master-cert
+  subject:
+    organizations:
+    - node-feature-discovery
+  commonName: nfd-master
+  dnsNames:
+  # must match the service name
+  - {{ include "node-feature-discovery.fullname" . }}-master
+  # first one is configured for use by the worker; below are for completeness
```
Comment (marquiz) [inline deployment/helm/node-feature-discovery/templates/cert-manager-certs.yaml:16]: nit: trailing whitespace

```suggestion
  # first one is configured for use by the worker; below are for completeness
```

File: deployment/helm/node-feature-discovery/values.yaml [hunk 3]
```diff
@@ -429,3 +429,13 @@ rbac:
   ## Annotations for the Service Account
   ##
   serviceAccountAnnotations: {}
+
+# Optionally use encryption for worker <--> master comms
+# TODO: verify hostname is not yet supported
+#
```
Comment (marquiz) [inline deployment/helm/node-feature-discovery/values.yaml:435]: nit: trailing whitespace

```suggestion
#
```

File: deployment/helm/node-feature-discovery/templates/cert-manager-certs.yaml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    name: nfd-ca-issuer
+    kind: Issuer
+    group: cert-manager.io
+
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: nfd-worker-cert
+  namespace: node-feature-discovery
```
Comment (marquiz) [inline deployment/helm/node-feature-discovery/templates/cert-manager-certs.yaml:31]: I think you need to use `{{ $.Release.Namespace }}` here, too. Ditto all the other yamls, too...

```suggestion
  namespace: {{ $.Release.Namespace }}
```

**Review by marquiz — COMMENTED** [review]
Looks good. Just few small comments. Almost ready to merge, I'd say

File: deployment/helm/node-feature-discovery/templates/topologyupdater.yaml (hunk 1 above)
Comment (dbaker-rh) [reply deployment/helm/node-feature-discovery/templates/topologyupdater.yaml]: Agree!

File: deployment/helm/node-feature-discovery/templates/cert-manager-certs.yaml (hunk 4 above)
Comment (dbaker-rh) [reply deployment/helm/node-feature-discovery/templates/cert-manager-certs.yaml]: I think I actually need to remove it entirely (which I've done); helm itself will manage the namespace it's being installed into.

Comment (marquiz) [reply deployment/helm/node-feature-discovery/templates/cert-manager-certs.yaml]: Uh-oh, true 👍 Sorry for my helm-noobness 🙄

**Review by marquiz — COMMENTED** [review]
One open nit but after that I think you could squash these into one commit and we could get this merged.

**Comment by dbaker-rh** [comment]
Qs:

* I have a documentation edit in the other (related, for kustomization templates) PR #710.  If that one isn't going to merge shortly, I'll suggest that I duplicate the edit in this one as well.
* What's the open nit?  I thought I knocked them all out (I notice I had cleaned up the single whitespace nit but not cleared it in the PR page, so perhaps it was that?)
* Were you able to test/verify the topology exporter updates are good?  I've not yet had chance to build a new test cluster with that feature gate enabled to test it myself.
*

**Comment by marquiz** [comment]
> * I have a documentation edit in the other (related, for kustomization templates) PR [Fix kustomization template to work with cert-manager #710](https://github.com/kubernetes-sigs/node-feature-discovery/pull/710).  If that one isn't going to merge shortly, I'll suggest that I duplicate the edit in this one as well.

I think we can merge #710 soon. There is basically this one nit that I'd like to get addressed.

> * What's the open nit?  I thought I knocked them all out (I notice I had cleaned up the single whitespace nit but not cleared it in the PR page, so perhaps it was that?)

Yeah, it was that. Seems to be fixed now 👌 

> * Were you able to test/verify the topology exporter updates are good?  I've not yet had chance to build a new test cluster with that feature gate enabled to test it myself.

Yeah, tested that quickly in the latest version and seems to be fine. Thanks!

File: docs/get-started/deployment-and-usage.md [hunk 5]
```diff
@@ -436,17 +438,14 @@ management between nfd-master and the nfd-worker pods.
...
 [CA Issuer](https://cert-manager.io/docs/configuration/ca/) to sign
 `Certificate` requests for NFD components in `node-feature-discovery`
 namespace.
 
 ```bash
-kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v1.5.1/cert-manager.yaml
-openssl genrsa -out deployment/overlays/samples/cert-manager/tls.key 2048
-openssl req -x509 -new -nodes -key deployment/overlays/samples/cert-manager/tls.key -subj "/CN=nfd-ca" \
-        -days 10000 -out deployment/overlays/samples/cert-manager/tls.crt
+kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v1.6.1/cert-manager.yaml
```
Comment (marquiz) [inline docs/get-started/deployment-and-usage.md:448]: nit: unrelated, this section is basically related to kustomize. This section should be modified to cover Helm too, though 🤔

Comment (dbaker-rh) [reply docs/get-started/deployment-and-usage.md:448]: Agree.  A general TLS page in the docs would be useful.

**Review by marquiz — APPROVED** [review]
Let's get this in
/lgtm
